### PR TITLE
[release/9.0-preview2] Revert "Support generic arguments for calli in CoreCLR (#97079)"

### PR DESCRIPTION
**Pull request by jkotas**
This reverts commit 15ff723dfe6b0fc90797c8b3117b29e51bcddab0.

Revert of #97079 in release/9.0-preview2

/cc @AaronRobinsonMSFT 

## Customer Impact

- [ ] Customer reported
- [x] Found internally

## Regression

- [x] Yes
- [ ] No

Regression introduced by #97079. Breaks all CsWinRT scenarios.

## Testing

New tests addressing the scenario will be added in main.

## Risk

Low. Full revert of the change #97079

### CODE REVIEW

**Comment by jkotas** [comment]
@carlossanlop Is there an approval process that this needs to go through?

This needs to be reverted in preview2. It breaks all CsWinRT scenarios.

**Comment by carlossanlop** [comment]
Yes, we need Tactics approval for this.

Please fill out the template, add the `servicing-consider` label, send the email to Tactics requesting approval, and get the PR signed-off. I'll monitor it so that when it's ready, I'll merge it.

**Review by carlossanlop — APPROVED** [review]
Approving as it's a revert. Just need to make sure the CI looks good before merging.

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Pull request merged**
